com.unity.shadergraph/Editor/Drawing/Views/MaterialGraphView.cs

```diff
@@ -323,6 +331,57 @@ internal static void InsertCopyPasteGraph(this MaterialGraphView graphView, Copy
             if (copyGraph == null)
                 return;
 
+            foreach (IShaderProperty property in copyGraph.properties)
+            {
+                string propertyName = graphView.graph.SanitizePropertyName(property.displayName);
+
+                IShaderProperty copiedProperty = null;
+
+                if (property is Vector1ShaderProperty)
+                {
+                    copiedProperty = new Vector1ShaderProperty();
```

Comment:
Avoid the cast here by instantiating as a new variable within this scope as a `Vector1ShaderProperty`, then set `copiedProperty` to the variable after you've set `value`.

Reply:
Also, this should go into a method on `AbstractShaderProperty` that allows you to make a copy of a property (or copy from another one - decide on some way :p ), since property types likely have more information than just `value`.